mobile/lib/modules/backup/providers/backup.provider.dart

```diff
@@ -404,13 +482,88 @@ class BackupNotifier extends StateNotifier<BackUpState> {
         return;
       }
 
+      if (state.backupProgress == BackUpProgressEnum.inBackground) {
+        debugPrint("[resumeBackup] Background backup is running - abort");
+        return;
+      }
+
       // Run backup
       debugPrint("[resumeBackup] Start back up");
-      startBackupProcess();
+      await startBackupProcess();
     }
 
     return;
   }
+
+  Future<void> resumeBackup() async {
```

Comment:
I understand this is called from the external context of the file, let's put in some comments so late comers understand why there are two `resumeBackup` functions here